addons/mail/static/src/core_ui/message.js

```diff
@@ -315,15 +316,24 @@ export class Message extends Component {
     }
 
     onClickPin() {
-        const prompt = this.message.pinned_at
-            ? _t("Are you sure you want to remove this pinned message?")
-            : _t(
-                  "The following message will be pinned to the channel. Are you sure you want to continue?"
-              );
+        const thread = this.message.originThread;
         this.env.services.dialog.add(MessageConfirmDialog, {
+            confirmColor: this.message.pinned_at ? "btn-danger" : undefined,
+            confirmText: this.message.pinned_at
+                ? _t("Yes, remove it please")
+                : _t("Oh yeah. Pin it"),
             message: this.message,
             messageComponent: Message,
-            prompt,
+            prompt: this.message.pinned_at
+                ? _t("Enough of that, I guess. You sure you want to remove this pinned message?")
+                : sprintf(
+                      _t(
+                          "Hey, just double checking that you want to pin this message to %(conversation_name)s for posterity and greatness?"
+                      ),
+                      { conversation_name: thread.prefix + thread.displayName }
+                  ),
+            size: "md",
+            title: this.message.pinned_at ? _t("Unpin Message") : _t("Pin It. Pin It Good."),
```

Comment:
Have to say this reads like bad 18+ movie.

Reply:
Doesn't sound sexual in English, but it still sounds awkward... Makes me think of the 80s... https://youtu.be/IIEVqFB4WUo